### ISSUE TITLE
exclude Topic pattern

### ISSUE THREAD

**Issue by sumitbaurai** [uber/uReplicator#142]
Hi,

While trying to use the -patternToExcludeTopics flag, I entered the pattern : ^[0-9_]*$

-patternToExcludeTopics ^[0-9_]*$

This is not working and not excluding the topics with a number in their name. 

While trying to debug, I analysed  the AutoTopicWhitelistingManager.java class 
it has the following log statement

LOGGER.info("BlacklistedTopics={} and ExcludingPattern={}", _blacklistedTopics,
        _patternToExcludeTopics);

But the actual output of this which I see from the Controller logs is : 

[2018-07-25 17:18:56,482] INFO BlacklistedTopics=[__consumer_offsets, __confluent.support
.metrics] and ExcludingPattern=**^[0-9_]*$|__consumer_offsets** (com.uber.stream.kafka.mirror
maker.controller.core.AutoTopicWhitelistingManager:155)

Somehow |__consumer_offsets is getting appended to the regex that I had provided OR am i seeing something wrong. 

Please help !!

**Comment by xhl1988** [comment]
__consumer_offsets is enforced to be appended here: https://github.com/uber/uReplicator/blob/master/uReplicator-Controller/src/main/java/com/uber/stream/kafka/mirrormaker/controller/ControllerConf.java#L165 since it is 'unconsumable'.

**Comment by sumitbaurai** [comment]
Hi, 
Not sure how does the pattern exclusion works. 

I configured the exclusion pattern but it does not work. My pattern is .*\d+.* ( the asterisks are not appearing in the view here ... to exclude any topic with a numeric in name). From the logs, i can see that it does not blacklist the topics (there are a number of topics with numerics). I also ran a test to publish messages on the source cluster (on a topic with a numeric in name) and the messages are being replicated to the target cluster

[2018-07-26 10:19:41,327] INFO Now serving 12 topics in Kafka cluster: srcKafkaCluster (com.uber.stream.kafka.mirrormaker.controller.core.KafkaBrokerTopicObserver:159)
[2018-07-26 10:19:41,360] INFO Now serving 13 topics in Kafka cluster: destKafkaCluster (com.uber.stream.kafka.mirrormaker.controller.core.KafkaBrokerTopicObserver:159)
[2018-07-26 10:19:41,791] INFO Trying to run the validation job (com.uber.stream.kafka.mirrormaker.controller.validation.ValidationManager:79)
[2018-07-26 10:19:41,810] INFO Trying to run topic whitelisting job (com.uber.stream.kafka.mirrormaker.controller.core.AutoTopicWhitelistingManager:120)
[2018-07-26 10:19:41,816] INFO BlacklistedTopics=[__consumer_offsets, __confluent.support.metrics] and ExcludingPattern=.*\d+.*|__consumer_offsets (com.uber.stream.kafka.mirrormaker.controller.core.AutoTopicWhitelistingManager:155)
[2018-07-26 10:19:41,816] INFO Exclude topic=__confluent.support.metrics by blacklist (com.uber.stream.kafka.mirrormaker.controller.core.AutoTopicWhitelistingManager:162)
[2018-07-26 10:19:41,816] INFO Exclude topic=__consumer_offsets by blacklist (com.uber.stream.kafka.mirrormaker.controller.core.AutoTopicWhitelistingManager:162)
[2018-07-26 10:19:41,816] INFO Trying to whitelist topics: [] (com.uber.stream.kafka.mirrormaker.controller.core.AutoTopicWhitelistingManager:128)
